src/GitVersionCore/VersionConverters/GitVersionInfo/Templates/GitVersionInformation.fs

```diff
@@ -8,6 +8,22 @@
 // </auto-generated>
 //------------------------------------------------------------------------------
 
+#if NET20 || NET35 || NETCOREAPP1_0 || NETCOREAPP1_1 || NETSTANDARD1_0 || NETSTANDARD1_1 || NETSTANDARD1_2 || NETSTANDARD1_3 || NETSTANDARD1_4 || NETSTANDARD1_5 || NETSTANDARD1_6
```

Comment:
Shouldn't `SHOULD_DEFINE_EXCLUDE_FROM_CODE_COVERAGE` be defined in F# as well?

Reply:
There's no preprocessor instruction in F# that allows defining a constant (when you need one you have to pass it to the compiler with a `-d` argument). And the `SHOULD_DEFINE_EXCLUDE_FROM_CODE_COVERAGE` constant has no special meaning, it's just used to split the `#if` as I said above.

I agree this is not consistant. If you prefer, I can have everything look more like the F# version.

Reply:
I would prefer consistency. Hopefully, we can replace these `#if` statements with a check before the files are generated in the first place.

Reply:
>Hopefully, we can replace these #if statements with a check before the files are generated in the first place.

I suppose you are referring to a solution similar to what was suggested in the [issue](https://github.com/GitTools/GitVersion/issues/2330#issuecomment-645669350): having the support of the attribute detected by MSBuild? That would probably work but:
* I'm not that familiar (yet) with how MSBuild code works in GitVersion,
* Will you ever want to support generation of `GitVersionInformation` class from the command-line utility? If so, I think we can't rely on constants defined in a MSBuild file that would not exist in this case.

Anyway, I don't have clear ideas to how we can reliably get rid of this ugly `#if`, but in the meantime and if you're ok with this. I'll simply have everything look more like the F# version.

Reply:
> I suppose you are referring to a solution similar to what was suggested in the [issue](https://github.com/GitTools/GitVersion/issues/2330#issuecomment-645669350): having the support of the attribute detected by MSBuild? 

Correct. I agree that the solution is difficult, though, which is why I think we should go ahead with this `#if`-based solution first.

>Anyway, I don't have clear ideas to how we can reliably get rid of this ugly #if, but in the meantime and if you're ok with this. I'll simply have everything look more like the F# version.

👍